hngprojects/hng_boilerplate_java_web
# feat: implement status page endpoints to display status information

## Description

This PR implements a new Status Page module that integrates Newman for API testing and provides endpoints to retrieve API status information.

### Key changes:
- Created a new `statuspage` package to encapsulate all status page functionality
- Implemented Newman integration for running API tests periodically
- Added database persistence for API status results
- Created REST endpoints to retrieve API status information

### How to manually test:
1. Ensure you have Newman installed globally: `npm install -g newman`
2. Update `application.properties` with your Postman collection path, environment path, and desired output path
3. Run the application
4. Wait for the scheduled task to run (every 15 minutes) or manually trigger a test run
5. Access the endpoint `GET /api/status` to view the status results
